packages/core/test/application_init_spec.ts

```diff
@@ -73,5 +74,58 @@ import {inject, TestBed, waitForAsync} from '../testing';
            });
          })));
     });
+
+    describe('with app initializers represented using observables', () => {
+      let subscriber: Subscriber<any>;
+      let observable: Observable<any>;
+      let completerResolver = false;
+      beforeEach(() => {
+        observable = new Observable((res) => {
+          subscriber = res;
+        });
+        TestBed.configureTestingModule({
+          providers: [
+            {provide: APP_INITIALIZER, multi: true, useValue: () => observable},
+          ]
+        });
+      });
+
+      it('should update the status once all async observable initializers are done',
+         waitForAsync(inject([ApplicationInitStatus], (status: ApplicationInitStatus) => {
+           // Accessing internal `runInitializers` function of the `ApplicationInitStatus` class
+           // instance for testing purposes to invoke initializer functions.
+           (status as any).runInitializers();
+
+           setTimeout(() => {
+             completerResolver = true;
+             subscriber.next();
+             subscriber.complete();
+           });
+
+           expect(status.done).toBe(false);
+           status.donePromise.then(() => {
+             expect(status.done).toBe(true);
+             expect(completerResolver).toBe(true);
+           });
+         })));
+
+      it('should update the status once async observable initializers has error',
+         waitForAsync(inject([ApplicationInitStatus], (status: ApplicationInitStatus) => {
+           // Accessing internal `runInitializers` function of the `ApplicationInitStatus` class
+           // instance for testing purposes to invoke initializer functions.
+           (status as any).runInitializers();
+
+           setTimeout(() => {
+             completerResolver = true;
+             subscriber.error();
+           });
+
+           expect(status.done).toBe(false);
+           status.donePromise.catch(() => {
+             expect(status.done).toBe(false);
+             expect(completerResolver).toBe(true);
+           });
```

Comment:
I'd propose adding `then` here to make sure it's not invoked when observable emits an error:
```suggestion
           status.donePromise
             .then(() => fail('`donePromise.then` should not be invoked when observable emits an error'))
             .catch(() => {
               expect(status.done).toBe(false);
               expect(completerResolver).toBe(true);
             });
```